pkg/blob/runner.go

```diff
@@ -63,22 +64,22 @@ type batchRunner struct {
 	kernel.ServiceStage
 
 	logger   log.Logger
-	metric   metric.Writer
-	client   gosoS3.Client
 	channels *BatchRunnerChannels
+	client   gosoS3.Client
+	metric   metric.Writer
 	settings *BatchRunnerSettings
 }
 
-func NewBatchRunner(ctx context.Context, config cfg.Config, logger log.Logger) (*batchRunner, error) {
+func NewBatchRunner(ctx context.Context, config cfg.Config, logger log.Logger) (BatchRunner, error) {
 	settings := &BatchRunnerSettings{}
 	config.UnmarshalKey("blob", settings)
```

Comment:
i'd love to see something like 
```suggestion
func NewBatchRunner(ctx context.Context, config cfg.Config, logger log.Logger, name string) (BatchRunner, error) {
	settings := &BatchRunnerSettings{}
	config.UnmarshalKey(fmt.Sprint("blob.%s", name), settings)
```